core/src/test/scala/unit/kafka/utils/JsonTest.scala

```diff
@@ -40,25 +40,34 @@ class JsonTest {
   def testJsonParse(): Unit = {
     val jnf = JsonNodeFactory.instance
 
-    assertEquals(Json.parseFull("{}"), Some(JsonValue(new ObjectNode(jnf))))
+    assertEquals(Some(JsonValue(new ObjectNode(jnf))), Json.parseFull("{}"))
+    assertEquals(Right(JsonValue(new ObjectNode(jnf))), Json.tryParseFull("{}"))
+    assertThrows(classOf[IllegalArgumentException], () => Json.tryParseFull(null))
+    assertThrows(classOf[IllegalArgumentException], () => Json.tryParseBytes(null))
 
-    assertEquals(Json.parseFull("""{"foo":"bar"s}"""), None)
+    assertEquals(Option(MissingNode.getInstance()).map(JsonValue(_)), Json.parseFull(""))
+    assertEquals(Right(MissingNode.getInstance()).map(JsonValue(_)), Json.tryParseFull(""))
```

Comment:
Is this what we want? Or would we want this to `Left` and the line above to be `None`?

Reply:
You're right, it makes sense to return `None` and `Left` in those cases. Will change it and upload it to this PR.